addons/account/models/account_move.py

```diff
@@ -1683,17 +1714,43 @@ def _inverse_partner_id(self):
 
     @api.onchange('company_id')
     def _inverse_company_id(self):
+        self.exchange_rate = False
         self._conditional_add_to_compute('journal_id', lambda m: (
             not m.journal_id.filtered_domain(self.env['account.journal']._check_company_domain(m.company_id))
         ))
 
     @api.onchange('currency_id')
     def _inverse_currency_id(self):
+        self.exchange_rate = False
         (self.line_ids | self.invoice_line_ids)._conditional_add_to_compute('currency_id', lambda l: (
             l.move_id.is_invoice(True)
             and l.move_id.currency_id != l.currency_id
         ))
 
+    @api.onchange('display_rate')
+    def _inverse_display_rate(self):
+        res = {}
+        for move in self:
+            if move.display_rate and move.display_rate != move.system_exchange_rate:
+                move.exchange_rate = move.display_rate
+                if move._origin.display_rate and move.display_rate != move._origin.display_rate:
+                    diff = (move._origin.display_rate - move.display_rate) / move._origin.display_rate
+                    if abs(diff) > 0.2:
+                        res['warning'] = {
+                            'title': _("Warning for %s", self.currency_id.name),
+                            'message': _(
+                                "The new rate is quite far from the previous rate.\n"
+                                "Incorrect currency rates may cause critical problems, make sure the rate is correct!\n"
+                                "The previous rate was %s",
+                                self._origin.display_rate,
+                            )
+                        }
+            else:
+                move.exchange_rate = False
+
+        (self.line_ids | self.invoice_line_ids).invalidate_recordset(fnames=['currency_rate'])
```

Comment:
Doesn't adding the dependency on `move_id.exchange_rate` for `currency_rate` work?